src/Pos/Delegation/Types.hs

```diff
@@ -68,3 +69,9 @@ type DlgMemPool = HashMap PublicKey ProxySKHeavy
 
 -- | Confirmation of light cert type.
 type ProxySKLightConfirmation = (ProxySKLight, ProxySigLight ProxySKLight)
+
+-- | Lightweight PSK or heavyweight PSK with real leader public key
+-- (because heavyweight psks have redelegation feature, so pskIssuerPk
+-- hPsk /= leader in general case). This is used to create a block
+-- header only.
+type ProxySKBlockInfo = Either ProxySKLight (ProxySKHeavy, PublicKey)
```

Comment:
I wrote it in older commit and I will write again.
This type is **always** used with `Maybe`. This commit confirms it. And it makes sense, because to sign block this thing is indeed optional (we can use regular secret key or we can use PSK). And if we add `Maybe` here, it will be too cumbersome type which should be `data`.
This type denotes the way to sign a block. And there are three possibilities. So why use `Maybe $ Either bla (patak, kek)`? Why not `SignRegular | SignProxyLight ProxySKLight | SignProxyHeavy { sphPSK :: !ProxySKHeavy, sphIssuerKey :: !PublicKey }`?